dplutils/pipeline/stream.py

```diff
@@ -135,7 +135,9 @@ def process_source(self, source):
                 source_batch.append(next(self.source_generator))
             except StopIteration:
                 self.source_exhausted = True
-                return
+                if len(source_batch) == 0:
+                    return
+                break
             self.n_sourced += 1
             if self.n_sourced == self.max_batches:
```

Comment:
Yes, I see the confusion here. At the moment it is counting batches from the source and sourcing at most `max_batches` of them. The `batch_size` parameter of a task changes the batch itself, so it is not contradictory. But also handling of input batch size > 1 is not great. This is a separate issue though (just created https://github.com/ssec-jhu/dplutils/issues/76)

Reply:
No, it is technically counting the number of source batches processed. Do you think there is an issue related to this PR (didn't touch that part)? Otherwise I think confusion can be alleviated in separate change from issue 76